avalanche/training/templates/observation_type/batch_observation.py

```diff
@@ -22,36 +43,29 @@ def model_adaptation(self, model=None):
             # If the strategy has access to task boundaries, adapt the model
             # for the whole origin experience to add the
             if self.experience.access_task_boundaries:
-                avalanche_model_adaptation(model,
+                avalanche_model_adaptation(model, 
                                            self.experience.origin_experience)
             else:
                 avalanche_model_adaptation(model, self.experience)
-
-        # For evaluation, the experience is not necessarily an online
-        # experience:
         else:
             avalanche_model_adaptation(model, self.experience)
 
         return model.to(self.device)
 
-    def make_optimizer(self):
+    def make_optimizer(self, reset_opt=True, reset_state=False):
```

Comment:
since this are global arguments of the train method I would be more explicit `reset_optimizer`, `reset_optimizer_state`

Reply:
also, add a docstring for the arguments